File: projects/gmail-outreach/read-pe-sheet.py
import json
from google.oauth2 import service_account
from googleapiclient.discovery import build

# Sheet ID
SHEET_ID = '11TRs92xmRWJ_FEQ_0nnLDrUkPPJRSqTG_iBSYBjGov4'
SERVICE_ACCOUNT_FILE = 'service-account.json'

# Setup credentials
credentials = service_account.Credentials.from_service_account_file(
    SERVICE_ACCOUNT_FILE,
    scopes=['https://www.googleapis.com/auth/spreadsheets']
)

# Build service
service = build('sheets', 'v4', credentials=credentials)
sheet = service.spreadsheets()

# Read the sheet
result = sheet.values().get(
    spreadsheetId=SHEET_ID,
    range='Sheet1!A1:Z1000'
).execute()

values = result.get('values', [])

if not values:
    print('No data found.')
else:
    # Print header
    header = values[0] if values else []
    print(f"Found {len(values)-1} rows")
    print(f"Columns: {header}")
    
    # Save to JSON for analysis
    output = []
    for i, row in enumerate(values[1:], 2):  # Start from row 2
        if len(row) > 0:  # Skip empty rows
            lead = {}
            for j, col in enumerate(header):
                lead[col] = row[j] if j < len(row) else ''
            lead['_row'] = i
            output.append(lead)
    
    with open('current-pe-data.json', 'w', encoding='utf-8') as f:
        json.dump(output, f, indent=2, ensure_ascii=False)
    
    print(f"\nSaved {len(output)} leads to current-pe-data.json")
    
    # Count enrichment needs
    needs_enrichment = []
    for lead in output:
        contact = lead.get('Contact Name', '').strip()
        email = lead.get('Email', '').strip().lower()
        status = lead.get('Status', '').strip()
        
        # Check if needs enrichment
        if (not contact or 
            not email or 
            email.startswith('info@') or 
            email.startswith('sales@') or 
            email.startswith('ir@') or 
            email.startswith('contact@')):
            if status not in ['Dead', 'Invalid', 'Bounced', 'Sent']:
                needs_enrichment.append({
                    'row': lead['_row'],
                    'company': lead.get('Company', ''),
                    'contact': contact,
                    'email': email,
                    'status': status
                })
    
    print(f"\n{len(needs_enrichment)} leads need enrichment:")
    for i, lead in enumerate(needs_enrichment[:15], 1):
        print(f"{i}. Row {lead['row']}: {lead['company']} - {lead['contact'] or '(no contact)'} - {lead['email'] or '(no email)'}")

File: read-pe-sheet.js
const { google } = require('googleapis');
const fs = require('fs');

async function readSheet() {
  try {
    const auth = new google.auth.GoogleAuth({
      keyFile: './projects/gmail-outreach/service-account.json',
      scopes: ['https://www.googleapis.com/auth/spreadsheets'],
    });
    
    const sheets = google.sheets({ version: 'v4', auth });
    const spreadsheetId = '11TRs92xmRWJ_FEQ_0nnLDrUkPPJRSqTG_iBSYBjGov4';
    
    const result = await sheets.spreadsheets.values.get({
      spreadsheetId,
      range: 'Sheet1!A:N',
    });
    
    const rows = result.data.values || [];
    console.log(`Total rows: ${rows.length}`);
    
    if (rows.length > 0) {
      const headers = rows[0];
      console.log('\nHeaders:', headers);
      
      // Find leads needing enrichment (empty Contact Name or generic email)
      const needsEnrichment = [];
      
      for (let i = 1; i < rows.length; i++) {
        const row = rows[i];
        const company = row[0] || '';
        const contactName = row[2] || '';
        const email = row[4] || '';
        const status = row[9] || '';
        
        // Skip if already enriched or contacted
        if (status.toLowerCase().includes('contact') || status.toLowerCase() === 'enriched') {
          continue;
        }
        
        // Check if needs enrichment
        const hasEmptyContact = !contactName || contactName.trim() === '';
        const hasGenericEmail = email && (
          email.toLowerCase().includes('info@') ||
          email.toLowerCase().includes('sales@') ||
          email.toLowerCase().includes('ir@') ||
          email.toLowerCase().includes('contact@')
        );
        
        if (hasEmptyContact || hasGenericEmail) {
          needsEnrichment.push({
            row: i + 1,
            company,
            contactName,
            email,
            status,
            website: row[5] || '',
            reason: hasEmptyContact ? 'Empty contact name' : 'Generic email'
          });
        }
      }
      
      console.log(`\nLeads needing enrichment: ${needsEnrichment.length}`);
      console.log('\nFirst 15 leads to enrich:');
      needsEnrichment.slice(0, 15).forEach(lead => {
        console.log(`${lead.company} | ${lead.contactName || '(empty)'} | ${lead.email || '(empty)'} | ${lead.reason}`);
      });
      
      // Save to file for further processing
      fs.writeFileSync('./projects/gmail-outreach/enrichment-targets-cron.json', JSON.stringify(needsEnrichment, null, 2));
    }
  } catch (error) {
    console.error('Error:', error.message);
  }
}

readSheet();
